photon-ml/src/main/scala/com/linkedin/photon/ml/algorithm/CoordinateDescent.scala

```diff
@@ -177,36 +188,48 @@ class CoordinateDescent(
         logger.info(s"Training objective function value after updating coordinate with id $coordinateId at " +
             s"iteration $iteration is:\n$objectiveFunctionValue")
 
-        // Update the validating score and evaluate the updated model on the validating data
-        validatingScoresContainerOption = validatingDataAndEvaluatorsOption.map { case (validatingData, evaluators) =>
+        // Update the validation score and evaluate the updated model on the validation data
+        // if there are evaluators defined
+        val currentEval = validatingDataAndEvaluatorsOption.map { case (validatingData, evaluators) =>
           var validatingScoresContainer = validatingScoresContainerOption.get
-          val validatingScores = updatedModel
-            .score(validatingData)
-            .setName(s"Updated validating scores with coordinateId $coordinateId")
-            .persistRDD(StorageLevel.INFREQUENT_REUSE_RDD_STORAGE_LEVEL)
-            .materialize()
+          validatingScoresContainer = validatingScoresContainer.updated(coordinateId,
+            updatedModel
+              .score(validatingData)
+              .setName(s"Updated validating scores for coordinateId $coordinateId")
+              .persistRDD(StorageLevel.INFREQUENT_REUSE_RDD_STORAGE_LEVEL)
+              .materialize()
+          )
           validatingScoresContainer(coordinateId).unpersistRDD()
-          validatingScoresContainer = validatingScoresContainer.updated(coordinateId, validatingScores)
           val fullScore = validatingScoresContainer.values.reduce(_ + _)
-          evaluators.foreach { evaluator =>
+          evaluators.map { evaluator =>
             val validationTimer = Timer.start()
             val evaluationMetric = evaluator.evaluate(fullScore.scores)
-            logger.debug(s"Finished validating the model, time elapsed: ${validationTimer.stop().durationSeconds} (s).")
+            logger.debug(s"Finished model validation, time elapsed: ${validationTimer.stop().durationSeconds} (s).")
```

Comment:
More descriptive is usually good. But you seem to mention "more detailed" (with new timers) rather than "more descriptive"?